### ISSUE TITLE
Request parameters should have name

### ISSUE THREAD

**Issue by zuevmaxim** [ktorio/ktor#1868]
**Ktor Version and Engine Used (client or server and name)**
io.ktor:ktor-server-cio:1.3.2

**Describe the bug**
Missing header name should fail(see https://tools.ietf.org/html/rfc2616#section-4.2)

**To Reproduce**
Run code:
```
import io.ktor.http.cio.parseRequest
import io.ktor.utils.io.ByteReadChannel
import kotlinx.coroutines.runBlocking

fun main() {
    test(
        """
        GET / HTTP/1.1
        Host: www.example.com
        : value


        """.trimIndent()
    )
}

fun test(http: String) {
    val request = runBlocking { parseRequest(ByteReadChannel(http)) }!!
    println("method=${request.method};version=${request.version};uri={${request.uri}};headers=[${request.headers}]")
}

```

**Expected behavior**
An error expected - exception or null result.

**Comment by zuevmaxim** [comment]
The same problem with `parseResponce`:
```
import io.ktor.http.cio.parseResponse
import io.ktor.utils.io.ByteReadChannel
import kotlinx.coroutines.runBlocking

fun main() {
    test(
        """
        HTTP/1.1 200 OK
        : encoding


    """.trimIndent()
    )
}

fun test(http: String) {
    val response = runBlocking { parseResponse(ByteReadChannel(http)) }!!
    println("status=${response.status};statusText={${response.statusText}};headers=[${response.headers}];version=${response.version}")
}
```

**Comment by oleg-larshin** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/KTOR-378">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.

**Comment by vladf2n** [comment]
Hi! I'll fix this.

**Comment by e5l** [comment]
Fixed in 1.5.1
